context.go

```diff
@@ -445,3 +385,12 @@ func normalizeFlags(flags []Flag, set *flag.FlagSet) error {
 	}
 	return nil
 }
+
+func makeFlagNameVisitor(names *[]string) func(*flag.Flag) {
+	return func(f *flag.Flag) {
+		name := strings.Split(f.Name, ",")[0]
+		if name != "help" && name != "version" {
```

Comment:
I dunno!  This behavior came along for the ride :smile_cat: https://github.com/codegangsta/cli/blob/3a52a3b8124e194986964c547054f0a7b8f56892/context.go#L202-L224